src/main.cc

```diff
@@ -1781,6 +1787,18 @@ Parse(TRITONSERVER_ServerOptions** server_options, int argc, char** argv)
     return false;
   }
 
+  // Check if there is a conflict between --disable-auto-complete-config
+  // and --strict-model-config
+  if (disable_auto_complete_config && strict_model_config_present) {
+    if (!strict_model_config) {
+      std::cerr
+          << "Warning: Overriding deprecated '--strict-model-config' from "
+             "False to True in favor of '--disable-auto-complete-config'!"
+          << std::endl;
+    }
+    strict_model_config = true;
```

Comment:
Do we need a warning for when the `strict_model_config=true` and `disable_auto_complete_config` is false?

Reply:
In this case, the user gets the warning that they are using an deprecated CLI option from the switch statement. We should still allow them to use the `--strict-model-config` option to not break backward compatibility.